lib/IRGen/GenDecl.cpp

```diff
@@ -1394,16 +1403,16 @@ llvm::Function *LinkInfo::createFunction(IRGenModule &IGM,
     existing->setName(getName() + ".unique");
   }
 
-  llvm::Function *fn
-    = llvm::Function::Create(fnType, getLinkage(), getName());
-  if (insertBefore) {
-    IGM.Module.getFunctionList().insert(insertBefore->getIterator(), fn);
-  } else {
-    IGM.Module.getFunctionList().push_back(fn);
-  }
+  llvm::Function *fn = llvm::Function::Create(fnType, getLinkage(), getName());
   fn->setVisibility(getVisibility());
+  fn->setDLLStorageClass(getDLLStorage());
   fn->setCallingConv(cc);
 
+  if (insertBefore)
+    IGM.Module.getFunctionList().insert(insertBefore->getIterator(), fn);
+  else
+    IGM.Module.getFunctionList().push_back(fn);
+
```

Comment:
That was to have the visibility/calling convention together.  If you would rather that they just be split up around the original position of this, I can do that.